lib/bundler/plugin/dsl.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Bundler
+  # Dsl to parse the Gemfile looking for plugins to install
+  class Plugin::Dsl < Bundler::Dsl
+    alias_method :_gem, :gem # To use for plugin installation as gem
+
+    # So that we don't have to override all there methods to dummy ones
+    # explicitly.
+    # They will be handled by method_missing
+    [:gemspec, :gem, :path, :install_if, :platforms, :env].each {|m| undef_method m }
+
+    def initialize
+      super
+      @sources = Plugin::SourceList.new
+    end
+
+    def plugin(name, *args)
+      _gem(name, *args)
+    end
+
+    def method_missing(name, *args)
+      # Dummy evaluation
```

Comment:
can we please only ignore methods defined on `Bundler::Dsl` ? otherwise this could lead to bugs in this class